GrizzFit/src/components/WorkoutsListItem.js

```diff
@@ -0,0 +1,88 @@
+import React, { useContext } from 'react'
+import { View, Text, TouchableOpacity, StyleSheet } from 'react-native'
+import { MaterialIcons } from '@expo/vector-icons';
+import { darkModePalette } from '../styles/DarkModePalette';
+import { WorkoutExercisesContext } from '../providers/WorkoutExercisesProvider';
+import { MainScreenContext } from '../providers/MainScreenProvider';
+
+export default function WorkoutsListItem({ item, navigation }) {
+    const { removeWorkout } = useContext(WorkoutExercisesContext)
+    const { numberOfCreatedWorkouts, setNumberOfCreatedWorkouts } = useContext(MainScreenContext)
+
+    const handleRemove = () => {
+        removeWorkout(item);
+        setNumberOfCreatedWorkouts( numberOfCreatedWorkouts - 1 )
+    }
+
+    return (
+        <View style={styles.button}>
+            <View style={styles.workoutInfo}>
+                {item.isFavorite ?
+                    <TouchableOpacity onPress={() => console.log('Unfavoriting ' + item.title)}>
+                        <MaterialIcons
+                            name="star"
+                            size={30}
+                            color={darkModePalette.primary}
+                        />
+                    </TouchableOpacity>
+                    :
+                    <TouchableOpacity onPress={() => console.log('Favoriting ' + item.title)}>
+                        <MaterialIcons
+                            name="star-border"
+                            size={30}
+                            color={darkModePalette.white}
+                        />
+                    </TouchableOpacity>
+                }
+                <TouchableOpacity onPress={() => navigation.push('WorkoutInfo', { workout: item })}>
+                    <Text style={styles.buttonText}>
+                        {item.title}
+                    </Text>
+                    <Text style={styles.buttonMiniText}>
+                        {item.tags}
+                    </Text>
+                </TouchableOpacity>
+            </View>
+            <TouchableOpacity onPress={handleRemove}>
+                <MaterialIcons
+                    name="close"
+                    size={24}
+                    color={darkModePalette.red}
```

Comment:
sus

Reply:
i mean it's not that sus i guess. it just makes me nervous thats all.

Reply:
understandably sus